File: Matchismo/Objective-C/Matchismo/Matchismo/Model/Card.m
//
//  Card.m
//  Matchismo
//
//  Created by Afonso Graça on 11/10/14.
//  Copyright (c) 2014 Afonso Graça. All rights reserved.
//
#import "Card.h"

@implementation Card

- (int)match:(NSArray *)otherCards {
	int score = 0;
	for (Card *card in otherCards) {
		if ([card.contents isEqualToString:self.contents]) {
			score = 1;
		}
	}
	return score;
}

@end

File: Matchismo/Swift/Matchismo/Matchismo/Model/Card.swift
//
//  Card.swift
//  Matchismo
//
//  Created by Afonso Graça on 12/10/14.
//  Copyright (c) 2014 Afonso Graça. All rights reserved.
//

import Foundation

class Card : NSObject {
	var chosen = false, matched = false
	var contents : String { return "" }

	func match(otherCards : [Card]) -> Int {
		var score = 0
		for card in otherCards {
			if card.contents == self.contents {
				score = 1
			}
		}
		return score
	}
}